tough/src/schema/spki.rs

```diff
@@ -67,14 +68,24 @@ pub(super) fn decode(
                     der::expect_tag_and_get_value(input, der::Tag::Sequence).and_then(
                         |alg_ident| {
                             alg_ident.read_all(ring::error::Unspecified, |input| {
-                                if der::expect_tag_and_get_value(input, der::Tag::OID)?
-                                    != untrusted::Input::from(&asn1_encode_oid(algorithm_oid))
+                                let expected_tag_value =
+                                    der::expect_tag_and_get_value(input, der::Tag::OID)?;
+
+                                let asn1_encode = asn1_encode_oid(algorithm_oid);
+                                let algo_encode_oid: Input<'_> =
+                                    untrusted::Input::from(&asn1_encode);
+                                if expected_tag_value.as_slice_less_safe()
```

Comment:
For posterity, I was concerned about using functions with `_less_safe` in the name. It seems that the `untrusted::Input` type tends to push you towards using a `Reader` and explicitly doing a byte-for-byte comparison as an acknowledgment that the comparison uses a boundless amount of compute. It also prevents you from parsing the input more than once.

It's easy for us in this context to see that we aren't parsing multiple times, we're just doing a bit comparison. For the problem of unbounded inputs, I suppose here we're always bounded by the size of `algorithm_oid`.

It might be good to add commentary about why this is safe, it makes me a bit squirrely to see code that implies that it could panic.

Reply:
Yeah, that's fair. I'll add a comment since I do think being "less safe" here is actually safe for us.